site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,316 @@
+  "auth-id": "${authentication_identifier}"
+}
+~~~
+
+The `type` property MUST contain the exact type of credentials to look up. The `auth-id` property MUST contain the authentication identifier to look up credentials for.
+
+**Response Message Format**
+
+A response to a *get credentials* request only contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message includes payload as defined in [Response Payload]({{< relref "#response-payload" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *200* | OK, the payload contains the credentials for the authentication identifier. |
+| *404* | Not Found, there are no credentials registered matching the critera. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Update Credentials
+
+Clients use this command to *update* existing credentials registered for a device. All of the information that has been previously registered for the device gets *replaced* with the information contained in the request message.
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *update credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type           | Description |
+| :--------------- | :-------: | :----------------------- | :------------- | :---------- |
+| *device_id*      | yes       | *application-properties* | UTF-8 *string* | MUST contain the ID of the device. |
+| *action*         | yes       | *application-properties* | UTF-8 *string* | MUST contain the value `update`. |
+
+The request message MUST contain payload as defined in [Request Payload]({{< relref "#request-payload" >}}).
+
+**Response Message Format**
+
+A response to an *update credentials* request only contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's body is empty. 
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | No Content, the credentials have been updated successfully. |
+| *404* | Not Found, there are no credentials registered matching the critera. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Remove Credentials
+
+Clients use this command to *remove* credentials registered for a device. Once the credentials are removed, the device may no longer be able to authenticate with protocol adapters using the authentication mechanism that the removed credentials corresponded to.
```

Comment:
I'm thinking about the possibility to specify only "type" for deleting all credentials of same type, if it make sense and using type = * as action for removing all credentials for the device to avoid another operation.